rust/template/distributed_datalog/src/schema/config.rs

```diff
@@ -0,0 +1,140 @@
+use std::cmp::Ordering;
+use std::collections::BTreeMap;
+use std::collections::BTreeSet;
+use std::path::PathBuf;
+
+use serde::Deserialize;
+use serde::Serialize;
+
+use differential_datalog::program::RelId;
+
+use crate::schema::Relations;
+
+/// All the input sources we support.
+#[derive(Clone, Debug, Eq, PartialEq, Deserialize, Serialize)]
+pub enum Source {
+    /// Input is coming from a file.
+    File(PathBuf),
+    // /// An observable directly used as a source.
+    // ///
+    // /// This variant is only meant for testing purposes.
+    // #[cfg(test)]
+    // Observable(observable: Box<dyn Observable>),
+}
+
+impl Ord for Source {
```

Comment:
Does Rust refuse to derive those automatically?

Reply:
Good point, I haven't actually tried it here. It would make sense if it just worked.